src/components/Card/Card.tsx

```diff
@@ -11,18 +12,41 @@ function Card({
   flip: boolean;
   onClick: () => void;
 }) {
+  const ref = React.useRef<HTMLDivElement>(document.createElement("div"));
```

Comment:
Here, you create an abstract div and access it before the div you actually need is initialized.

refs are usually initialized as `null`. Then you need to check that `ref.current` is not `null`, like `if (ref.current) { ... }`

Reply:
That was actually how I first approached it, but I still get TS warnings in callbacks. This can be solved by adding the if statement to each of them, but seems verbose, so I chose this solution instead.
<img width="615" alt="Снимок экрана 2023-06-14 в 19 23 10" src="https://github.com/rmxr/memo/assets/117208616/e07ffb03-6f90-4c86-9965-463d6af46b50">

Reply:
TS errors on the screenshot, I think, because `ref.current` comes from the outer scope. TS cannot be sure that `ref.current` won't be changed by the time it is used inside callbacks. Thus, if statement didn't help.
Assigning it to a variable solves the issue - now we're sure that going forward `div` holds the exact same value in later uses. And after `if (!div) { return }` it may not be `null`.